### DISCUSSION TITLE
How can we setup quorum environment for production for private and public transaction?

### DISCUSSION THREAD

**Discussion by ghost** [ConsenSys/quorum#1561] [Q&A] [answered]
Hi, I have to setup a production based environment where we can do private and public transactions. So please suggest me the best approach. Thanks in advanced!

**Comment by baptiste-b-pegasys** [comment, marked as answer]
Have you took a look at QBS? It's managed on Azure.
- https://azuremarketplace.microsoft.com/fr-fr/marketplace/apps/consensys.quorum-blockchain-service?tab=Overview
- https://consensys.net/quorum/qbs/

There is quorum on kubernetes and guidelines.
- https://consensys.net/docs/goquorum/en/latest/configure-and-manage/configure/bootnodes/#production-networks
- https://consensys.net/docs/goquorum/en/latest/tutorials/kubernetes/production/
- https://consensys.net/docs/goquorum/en/latest/configure-and-manage/configure/high-availability/

**Reply by ghost** [reply]
Thanks for your reply! As of now I don't want to use K8S So how can we setup multi host environment on 2 or 3 different VMs?
 
Kindly share any link (bootstrap project) or steps for this kind of setup.

Thanks in advance!!

**Reply by baptiste-b-pegasys** [reply]
I think you should start with https://consensys.net/docs/goquorum/en/latest/tutorials/private-network/create-qbft-network/
so you will have generated keys for your nodes, and instead of running geth locally, in different VMs, you will have the same geth arguments with the different keys.
staticnodes and permission files should use the IPs of your VMs, so some preparation in order to have 4 VMs with their IPs to share.

In quorum example, there is a vagrant file https://github.com/ConsenSys/quorum-examples/tree/master/vagrant
you could also overwrite the keys before. I never tested that.
https://github.com/ConsenSys/quorum-examples#running-with-vagrant

**Reply by ghost** [reply]
I am using "quorum-dev-quickstart" and running ./run.sh and all container are running with healthy status but Tessera is not running and getting below error in log file:

Please help me to resolve this issue!!!
/quorum-test-network-up/network/logs/tessera ---

{"timestamp":"2022-12-22 14:30:33,130","container":"01f2023e1ab9","level":"WARN","thread":"main","class":"e.l.all","logger":"eclipselink.logging.all","message":"Exception [EclipseLink-4002] (Eclipse Persistence Services - 3.0.2.v202107160933): org.eclipse.persistence.exceptions.DatabaseExceptionInternal Exception: org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "ST_AFFECTED_TRANSACTION" not found; SQL statement:ALTER TABLE ST_AFFECTED_TRANSACTION DROP CONSTRAINT FK_ST_AFFECTED_TRANSACTION_TXN_ID [42102-212]Error Code: 42102Call: ALTER TABLE ST_AFFECTED_TRANSACTION DROP CONSTRAINT FK_ST_AFFECTED_TRANSACTION_TXN_IDQuery: DataModifyQuery(sql="ALTER TABLE ST_AFFECTED_TRANSACTION DROP CONSTRAINT FK_ST_AFFECTED_TRANSACTION_TXN_ID")","throwable":""}
